lib/wrap.js

```diff
@@ -67,4 +68,4 @@ if (typeof mod === 'object' && mod.name) {
 }
 
 // Execute the wrapped script
-ext === 'mjs' ? import(main) : require(main);
+ext === 'mjs' || getPackageType.sync(main) === 'module' ? import(main) : require(main);
```

Comment:
I'd like to avoid blocking file operations, let's make method called isESM that returns a Promise which resolves a boolean and use that in both places where we need to check. This spot here is easy to handle since it's the end of the file.

Reply:
`getPackageType(main)` is async and returns a promise. I first tried to use that but I don't know how to do this without failing the tests. I get a lot of `test unfinished` problems… I can't simply place an await there because it's global code and we don't have top-level await. If I wrap it in a async lambda and call it, then it seems to exit too early…

Reply:
But why is sync so bad? It's just the start of the whole thing, right?

Reply:
sync does not scale well, you will only have one file operation happening at any given time instead of multiple happening simultaneously.

I was thinking too much about async imports and forgot that require is always synchronous so it actually doesn't matter. Please disregard what I said 😭.

Reply:
Yes I'm aware of sync not scaling well, but this only happens once per restart of the process, so I didn't think it to be a problem :) Glad we don't have to find a solution for the async code 🎉